quarkus:dependency-tree fails on conditional dependencies unless they are present in the local repo

### Describe the bug

Assuming the local repo does not contain the `io.quarkiverse.logging.logback` artifacts, create a project using `qs create -x logback -S 2.1` then

````
[aloubyansky@lenora code-with-quarkus]$ mvn quarkus:dependency-tree
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------< org.acme:code-with-quarkus >---------------------
[INFO] Building code-with-quarkus 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- quarkus-maven-plugin:2.1.0.CR1:dependency-tree (default-cli) @ code-with-quarkus ---
[INFO] Quarkus application PROD mode build dependency tree:
Downloading from central: https://repo.maven.apache.org/maven2/io/quarkiverse/logging/logback/quarkus-logging-logback/0.1.0/quarkus-logging-logback-0.1.0.pom
Downloaded from central: https://repo.maven.apache.org/maven2/io/quarkiverse/logging/logback/quarkus-logging-logback/0.1.0/quarkus-logging-logback-0.1.0.pom (1.9 kB at 3.9 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/io/quarkiverse/logging/logback/quarkus-logging-logback-parent/0.1.0/quarkus-logging-logback-parent-0.1.0.pom
Downloaded from central: https://repo.maven.apache.org/maven2/io/quarkiverse/logging/logback/quarkus-logging-logback-parent/0.1.0/quarkus-logging-logback-parent-0.1.0.pom (3.4 kB at 70 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/io/quarkiverse/logging/logback/quarkus-logging-logback/0.1.0/quarkus-logging-logback-0.1.0.jar
Downloaded from central: https://repo.maven.apache.org/maven2/io/quarkiverse/logging/logback/quarkus-logging-logback/0.1.0/quarkus-logging-logback-0.1.0.jar (3.1 kB at 64 kB/s)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.176 s
[INFO] Finished at: 2021-07-23T07:48:55+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.1.0.CR1:dependency-tree (default-cli) on project code-with-quarkus: Failed to resolve application model org.acme:code-with-quarkus::pom:1.0.0-SNAPSHOT dependencies: io.quarkus.bootstrap.resolver.maven.BootstrapMavenException: Failed to resolve artifact io.quarkiverse.logging.logback:quarkus-logging-logback-impl:jar:0.1.0: Could not find artifact io.quarkiverse.logging.logback:quarkus-logging-logback-impl:jar:0.1.0 -> [Help 1]
````
Building the project properly downloads the dependencies and then `quarkus:dependency-tree` works.

--- a/devtools/maven/src/main/java/io/quarkus/maven/DependencyTreeMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/DependencyTreeMojo.java
@@ -30,7 +30,7 @@ public class DependencyTreeMojo extends AbstractMojo {
     @Parameter(defaultValue = "${project}", readonly = true, required = true)
     protected MavenProject project;
 
-    @Parameter(defaultValue = "${project.remoteRepositories}", readonly = true, required = true)
+    @Parameter(defaultValue = "${project.remoteProjectRepositories}", readonly = true, required = true)
     private List<RemoteRepository> repos;
 
     /**

